shell/platform/darwin/ios/framework/Source/profiler_metrics_ios.mm

```diff
@@ -76,4 +76,33 @@
   return cpu_usage_info;
 }
 
+std::optional<MemoryUsageInfo> ProfilerMetricsIOS::MemoryUsage() {
+  kern_return_t kernel_return_code;
+  task_vm_info_data_t task_memory_info;
+  mach_msg_type_number_t task_memory_info_count = TASK_VM_INFO_COUNT;
+
+  kernel_return_code =
+      task_info(mach_task_self(), TASK_VM_INFO, reinterpret_cast<task_info_t>(&task_memory_info),
+                &task_memory_info_count);
+  if (kernel_return_code != KERN_SUCCESS) {
+    FML_LOG(ERROR) << " Error retrieving task memory information: "
+                   << mach_error_string(kernel_return_code);
+    return std::nullopt;
+  }
+
+  // `phys_footprint` is Apple's recommended way to measure app's memory usage. It provides the
+  // best approximate to xcode memory gauge. According to its source code explanation, the physical
+  // footprint mainly consists of app's internal memory data and IOKit mappings. `resident_size`
+  // is the total physical memory used by the app, so we simply do `resident_size - phys_footprint`
+  // to obtain the shared memory usage.
+  const double dirty_memory_usage =
+      static_cast<float>(task_memory_info.phys_footprint) / 1024.0 / 1024.0;
+  const double owned_shared_memory_usage =
+      static_cast<float>(task_memory_info.resident_size) / 1024.0 / 1024.0 - dirty_memory_usage;
```

Comment:
Is there any specific reason that it's casting to `float` first, and eventually becomes a `double`? Should both be `float` or `double`?

Reply:
You are right, as definition this should be both `double`.